C7/UIElements/Popups/PopupOverlay.cs

```diff
@@ -1,83 +1,106 @@
 using Godot;
-using ConvertCiv3Media;
-using System;
-using System.Diagnostics;
 using Serilog;
 
-public class PopupOverlay : HBoxContainer
+public partial class PopupOverlay : HBoxContainer
 {
 
 	private ILogger log = LogManager.ForContext<PopupOverlay>();
 
-	[Signal] public delegate void UnitDisbanded();
-	[Signal] public delegate void Quit();
-	[Signal] public delegate void BuildCity(string name);
-	[Signal] public delegate void HidePopup();
+	[Signal] public delegate void UnitDisbandedEventHandler();
+	[Signal] public delegate void QuitEventHandler();
+	[Signal] public delegate void BuildCityEventHandler(string name);
+	[Signal] public delegate void HidePopupEventHandler();
 
 	Control currentChild = null;
 
-	public const string NodePath = "/root/C7Game/CanvasLayer/PopupOverlay";
+	public static readonly string NodePath = "/root/C7Game/CanvasLayer/PopupOverlay";
+
+	private static readonly string ControlNodePath = "/root/C7Game/CanvasLayer/Control";
 
 	public enum PopupCategory {
 		Advisor,
 		Console,
 		Info	//Sounds similar to the above, but lower-pitched in the second half
 	}
 
-	public override void _Ready()
+	public void OnHidePopup()
 	{
-		base._Ready();
-	}
-
-	private void OnHidePopup()
-	{
-		log.Debug("Hiding popup");
+		// 1. enable mouse interaction with non-UI nodes
+		MouseFilter = MouseFilterEnum.Pass;
 		RemoveChild(currentChild);
+		currentChild = null;
 		Hide();
+
+		Control control = getControlNode();
+		// 2. enable mouse interactions with other UI elements
+		setMouseFilter(control, MouseFilterEnum.Pass);
+
+		// 3. enable clicking other UI elements
+		control.ProcessMode = ProcessModeEnum.Inherit;
 	}
 
-	public void PlaySound(AudioStreamSample wav)
+	public bool ShowingPopup => currentChild is not null;
+
+	public void PlaySound(AudioStreamWav wav)
 	{
 		AudioStreamPlayer player = GetNode<AudioStreamPlayer>("PopupSound");
 		player.Stream = wav;
 		player.Play();
 	}
 
+	private Control getControlNode() {
+		return GetNode<Control>(ControlNodePath);
+	}
+
+	private void setMouseFilter(Node n, MouseFilterEnum filter) {
+		foreach (Node child in n?.GetChildren()) {
+			setMouseFilter(child, filter);
+		}
+		if (n is Control control) {
+			control.MouseFilter = filter;
+		}
+	}
```

Comment:
let's make a TODO to look for a pattern for this. there has to be a better way than recursing through the whole scene

Reply:
#452